datafusion/physical-expr/src/aggregate/count_distinct.rs

```diff
@@ -152,6 +155,8 @@ impl AggregateExpr for DistinctCount {
             Float32 => float_distinct_count_accumulator!(Float32Type),
             Float64 => float_distinct_count_accumulator!(Float64Type),
 
+            Utf8 => Ok(Box::new(StringDistinctCountAccumulator::new())),
```

Comment:
`with capacities` is used in #8827, should we also need this, what should be the size for rawtable and buffer?

Reply:
I am not sure -- I need to look at memory allocation more carefully

Reply:
Update here is that we used `insert_accounted` -- that is not well documented but I think it is the correct API

We just need to write some tests to ensure the accounting is reasonable (like adding strings makes the memory accounting go up, and adding the same strings over and over does not)